src/layers/icmp.rs

```diff
@@ -59,6 +66,42 @@ pub struct IcmpUnsupported {
     unsupported: Vec<u8>,
 }
 
+#[derive(Default, Debug, Serialize)]
+#[serde(untagged)]
+pub enum IcmpData {
+    #[default]
+    None,
+    Unknown(IcmpUnknownData),
```

Comment:
nitpick: Probably better to call the Variants as follows - 

`Data(Data)`, `TimeStamp(TimestampData)`, `Error(OriginalDatagram)`